test/linkify-matrix-test.ts

```diff
@@ -0,0 +1,227 @@
+            expect(found).toEqual(([{
+                href: "#foo:bar.com",
+                type: "roomalias",
+                value: "#foo:bar.com",
+
+            }]));
+        });
+        it('accept #foo:com (mostly for (TLD|DOMAIN)+ mixing)', () => {
+            const test = '#foo:com';
+            const found = linkify.find(test);
+            expect(found).toEqual(([{
+                href: "#foo:com",
+                type: "roomalias",
+                value: "#foo:com",
+            }]));
+        });
+        it('accept repeated TLDs (e.g .org.uk)', () => {
+            const test = '#foo:bar.org.uk';
+            const found = linkify.find(test);
+            expect(found).toEqual(([{
+                href: "#foo:bar.org.uk",
+                type: "roomalias",
+                value: "#foo:bar.org.uk",
+            }]));
+        });
+        it('ignores trailing `:`', () => {
+            const test = '#foo:bar.com:';
+            const found = linkify.find(test);
+            expect(found).toEqual(([{
+                href: "#foo:bar.com",
+                type: "roomalias",
+                value: "#foo:bar.com",
+
+            }]));
+        });
+        it('accept :NUM (port specifier)', () => {
+            const test = '#foo:bar.com:2225';
+            const found = linkify.find(test);
+            expect(found).toEqual(([{
+                href: "#foo:bar.com:2225",
+                type: "roomalias",
+                value: "#foo:bar.com:2225",
+            }]));
+        });
+        it('ignores all the trailing :', () => {
+            const test = '#foo:bar.com::::';
+            const found = linkify.find(test);
+            expect(found).toEqual(([{
+                href: "#foo:bar.com",
+                type: "roomalias",
+                value: "#foo:bar.com",
+
+            }]));
+        });
+        it('properly parses room alias with dots in name', () => {
+            const test = '#foo.asdf:bar.com::::';
+            const found = linkify.find(test);
+            expect(found).toEqual(([{
+                href: "#foo.asdf:bar.com",
+                type: "roomalias",
+                value: "#foo.asdf:bar.com",
+
+            }]));
+        });
+        // This test does not work and does not parse the room
+        it.skip('does not parse room alias with too many separators', () => {
+            const test = '#foo:::bar.com';
+            const found = linkify.find(test);
+            expect(found).toEqual(([{
+                href: "#foo",
+                type: 'roomalias',
+                value: '#foo',
+            },
+            {
+                href: "http://bar.com",
+                type: "url",
+                value: "bar.com",
+            }]));
+        });
+        // This should not parse correctly, but it's been working this way in the previous version too
+        it.skip('does not parse multiple room aliases in one string', () => {
+            const test = '#foo:bar.com-baz.com';
```

Comment:
How is `#foo:bar.com-baz.com` multiple room aliases? `bar.com-baz.com` is a valid domain
subdomain=`bar`
domain=`com-baz`
tld=`com`

Reply:
Must have confused it with something else, there's been a lot of tests I did in the meantime and probably did some mess within rebases and commits. Fixed, thanks for catching!